js/style/style_layer.js

```diff
@@ -38,6 +38,8 @@ StyleLayer.prototype = {
                 }
                 this.layout['symbol-avoid-edges'] = true;
             }
+
+            this._resolvedLayout = new StyleDeclarationSet('layout', this.type, this.layout, this._constants || {});
```

Comment:
Maybe I just don't understand it well enough but I feel like our internal style API is turning into spaghetti code... This is :+1: for now but I'm daydreaming about refactoring...